app/routes.py

```diff
@@ -1,2 +1,272 @@
+def read_tasks():
+    tasks = Task.query.all()
+    response_body = []
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+        
+    for task in tasks:
+        response_body.append(
+            task.to_dict())
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_task(task_id):
+    check_not_found = is_parameter_found(Task, task_id)
+    if check_not_found:
+        return check_not_found
+    task = Task.query.get(task_id)
+    task_response = {}
+    task_response["task"] = task.to_dict()
+    return jsonify(task_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    check_not_found = is_parameter_found(Task, task_id)
+    if check_not_found:
+        return check_not_found
+
+    task = Task.query.get(task_id)
+    request_body = request.get_json()
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    db.session.commit()
+
+    response_body = {}
+    response_body["task"] = task.to_dict()
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PATCH"])
+def update_task_parameter(task_id):
+    check_not_found = is_parameter_found(Task, task_id)
+    if check_not_found:
+        return check_not_found
+
+    task = Task.query.get(task_id)
+    request_body = request.get_json()
+    if "title" in request_body:
+        task.name = request_body["title"]
+    if "description" in request_body:
+        task.description = request_body["description"]
+    
+    db.session.commit()
+    return make_response(f"Task {task_id} successfully updated!", 200)
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
+    check_not_found = is_parameter_found(Task, task_id)
+    if check_not_found:
+        return check_not_found
+
+    task = Task.query.get(task_id)
+    task.completed_at = datetime.now(timezone.utc)
+    response_body = {}
+    response_body = task.to_dict()
+
+    db.session.commit()
+    slack_url = "https://slack.com/api/chat.postMessage"
+    slack_bot_data = {"token": os.environ.get("ADA_BOT_TOKEN"),
+                        "channel": "task-notifications", 
+                        "text": (f"Someone just completed the task {task.title}") 
+                        }
+    requests.post(slack_url, slack_bot_data)
+    
+
+    return jsonify({"task": response_body}), 200
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incomplete(task_id):
+    check_not_found = is_parameter_found(Task, task_id)
+    if check_not_found:
+        return check_not_found
+
+    task = Task.query.get(task_id)
+    task.completed_at = None
+    response_body = {}
+    response_body = task.to_dict()
```

Comment:
We do not need to create an empty dictionary first.

```suggestion
    response_body = task.to_dict()
```